src/user-management/user-management.ts

```diff
@@ -294,6 +309,62 @@ export class UserManagement {
     return deserializeAuthenticationResponse(data);
   }
 
+  async authenticateWithSessionCookie({
+    sessionCookie,
+    cookiePassword = process.env.WORKOS_COOKIE_PASSWORD,
+  }: {
+    sessionCookie: string;
+    cookiePassword?: string;
+  }): Promise<
+    | AuthenticateWithSessionCookieSuccessResponse
+    | AuthenticateWithSessionCookieFailedResponse
+  > {
+    if (!cookiePassword) {
+      throw new Error('Cookie password is required');
+    }
+
+    if (!sessionCookie) {
+      return { authenticated: false };
+    }
+
+    const session = await unsealData<SessionCookieData>(sessionCookie, {
+      password: cookiePassword,
+    });
```

Comment:
`iron-session` provides a nice error when the cookie password is < 32 characters (also when sealing the cookie). Don't think we need additional checks here.

Reply:
We do the check in the Next/Remix libraries so it fails early if you have a too-short password set. Less important here since you're potentially passing in the password at runtime 👍